src/extension/options-page/Router/Contacts.tsx

```diff
@@ -1,38 +1,53 @@
-import React, { useMemo, useState } from 'react'
+import React, { useMemo, useState, Suspense, useRef, useTransition } from 'react'
 import DashboardRouterContainer from './Container'
-import { TextField, IconButton, Typography } from '@material-ui/core'
+import { TextField, IconButton, Typography, LinearProgress } from '@material-ui/core'
 import { makeStyles, createStyles } from '@material-ui/core/styles'
 import ClearIcon from '@material-ui/icons/Clear'
 import SearchIcon from '@material-ui/icons/Search'
-import { useFriendsList } from '../../../components/DataSource/useActivatedUI'
-import { ContactLine } from '../DashboardComponents/ContactLine'
+import { ContactLine, ContactLineSkeleton } from '../DashboardComponents/ContactLine'
 import { useI18N } from '../../../utils/i18n-next-ui'
 import { FixedSizeList } from 'react-window'
 import AutoResize from 'react-virtualized-auto-sizer'
+import { useSWRProfiles } from '../../../components/SuspenseDataSource/useSuspenseProfiles'
 
 const useStyles = makeStyles((theme) =>
     createStyles({
         title: {
             margin: theme.spacing(3, 0),
         },
+        progress: {
+            width: '1.5em',
+            height: '1.5em',
+            marginRight: '0.75em',
+        },
     }),
 )
 
+const contactLineFallback = (
+    <>
+        <ContactLineSkeleton />
```

Comment:
Generate it with an iterator.

Reply:
I think it's okay 🤔

Reply:
@neruthes Should skeleton fill the whole screen?